### PR TITLE
#9553: Improving readability of long attribute values in attribute table and table widgets

### PR DESCRIPTION
**Pull request by mahmoudadel54**
## Description
In this PR, I handled showing a tooltip for long attribute value in attribute table and table widgets only if user hovers on the cells that has shown content less than the attribute value. 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [x] Other... Please describe: Enhancement

## Issue
#9553 

**What is the current behavior?**
#9553 

**What is the new behavior?**
 Showing a tooltip for long attribute value in attribute table and table widgets only if user hovers on the cells that has shown content less than the attribute value. 

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
I created a generic enhancer with name "handleLongTextEnhancer" to handle the tooltip for long content

### CODE REVIEW

File: web/client/components/misc/enhancers/handleLongTextEnhancer.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+*
+* This source code is licensed under the BSD-style license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+import React from "react";
+import OverlayTrigger from "../OverlayTrigger";
+import { Tooltip } from "react-bootstrap";
+
+export  const handleLongTextEnhancer = (RenderFormatter) => (props) => {
```
Comment (MV88) [inline web/client/components/misc/enhancers/handleLongTextEnhancer.jsx:13]: maybe we can call this just Component or Wrapped
```suggestion
export  const handleLongTextEnhancer = (Component) => (props) => {
```

Comment (MV88) [inline web/client/components/misc/enhancers/handleLongTextEnhancer.jsx:13]: please add jsdoc here with also an example

File: web/client/components/misc/enhancers/handleLongTextEnhancer.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    const handleMouseEnter = () => {
+        const cellWidth = cellRef.current.offsetWidth;
+        const contentWidth = contentRef.current.offsetWidth;
+        setIsContentOverflowing(contentWidth > cellWidth);
+    };
+
+    return (<OverlayTrigger
+        placement="top"
+        overlay={isContentOverflowing ? <Tooltip id="tooltip">{RenderFormatter ? <RenderFormatter {...props} /> : value}</Tooltip> : <></>}
```
Comment (MV88) [inline web/client/components/misc/enhancers/handleLongTextEnhancer.jsx:27]: is the ternary needed? i mean don't we always have the Wrapped component when using this enhancer?

if it is not needed we can remove it here and in the below

**Review by MV88 — CHANGES_REQUESTED** [review]
looks ok, but I have some inline comments to get your attention

File: web/client/components/misc/enhancers/handleLongTextEnhancer.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import React from "react";
+import OverlayTrigger from "../OverlayTrigger";
+import { Tooltip } from "react-bootstrap";
+/**
+ * handleLongTextEnhancer enhancer. Enhances a long text content by adding a tooltip.
+ * @type {function}
+ * @name handleLongTextEnhancer
+ * @memberof components.misc.enhancers
+ * Wraps [wrapped component with content] to add tooltip for long content if shown content less than the main content
+ * @param {Component} Wrapped the wrapper to add tooltip for its long content
```
Comment (MV88) [inline web/client/components/misc/enhancers/handleLongTextEnhancer.jsx:18]:
```suggestion
 * @param {Component} Wrapped the component wrapped with a tooltip when its content is too long
```

**Review by MV88 — APPROVED** [review]

**Comment by MV88** [comment]
@ElenaGallo please test it in DEV

**Comment by ElenaGallo** [comment]
Test passed, @mahmoudadel54 please backport to 2023.02.xx. Thanks

**Comment by mahmoudadel54** [comment]
> Test passed, @mahmoudadel54 please backport to 2023.02.xx. Thanks

Backport is done --- > #9719

**Pull request merged**